### PR TITLE
Finish add user to subject and modify model

### PR DESCRIPTION
**Pull request by trungnn142**

### CODE REVIEW

File: app/models/course_subject.rb [hunk 1]
```diff
@@ -3,4 +3,18 @@ class CourseSubject < ActiveRecord::Base
...
+  after_destroy :destroy_subject_users
+
+  private
+  def add_subject_to_user
+    course.course_users.each do |course_user|
+      subjects.user_subjects.create user_id: course_user.user_id,
+        course_id: course_id
+    end
+  end
+  def destroy_subject_user
```
Comment (trinhducduy) [inline app/models/course_subject.rb]: new line

File: app/models/course_user.rb [hunk 2]
```diff
@@ -3,4 +3,18 @@ class CourseUser < ActiveRecord::Base
   belongs_to :user
 
   validates :course_id, uniqueness: {scope: :user_id}
+
+  after_save :add_user_to_subjects
+  after_destroy :destroy_user_subjects
+  private
```
Comment (trinhducduy) [inline app/models/course_user.rb]: new line

**Comment by ThuBM** [comment]
OK

File: app/models/course_subject.rb [hunk 3]
```diff
@@ -3,4 +3,19 @@ class CourseSubject < ActiveRecord::Base
...
+  private
+  def add_subject_to_user
+    course.course_users.each do |course_user|
+      subject.user_subjects.create user_id: course_user.user_id,
+        course_id: course_id
+    end
+  end
+
+  def destroy_subject_user
+    course.user_subjects.where(subject_id: subject_id).delete_all
```
Comment (dieunb) [inline app/models/course_subject.rb]: move `where` to scope

File: app/models/course_user.rb [hunk 4]
```diff
@@ -3,4 +3,19 @@ class CourseUser < ActiveRecord::Base
...
+  def add_user_to_subjects
+    course.course_subjects.each do |course_subject|
+      user.user_subjects.create subject_id: course_subject.subject_id,
+        course_id: course_id
+    end
+  end
+
+  def destroy_user_subjects
+    course.user_subjects.where(user_id: user_id).delete_all
```
Comment (dieunb) [inline app/models/course_user.rb]: scope

**Pull request merged**